BreastCancer/TasksAndSteps/APHExerciseCheckinTask.m

```diff
@@ -0,0 +1,560 @@
+                    NSInteger   index       = [textChoices indexOfObject:[answers firstObject]];
+                    
+                    //  Based on the selected answer lead the user to the correct step
+                    switch (index) {
+                        case 0:
+                        {
+                            stepIndex = [kReasonsTwoStepIndex integerValue];
+                            break;
+                        }
+                        default:
+                            break;
+                    }
+                }
+            }
+        }
+        
+        return stepIndex;
+    };
+    
+    NSInteger(^DetermineNextStep)(ORKStep*, ORKTaskResult*) = ^NSInteger(ORKStep* step, ORKTaskResult* result)
+    {
+        NSInteger stepIndex = 0;
+        
+        if (step != nil)
+        {
+            if (self.steps.count > 1)
+            {
+                //  Determine the level or depth of the step
+                NSInteger depth = DetermineStepDepth(step, result);
+                
+                //  Get the result, find the step identifier in steps corresponding to logical next step
+                //  Pass back the index.
+                switch (depth) {
+                    case 1:
+                    {
+                        //  Always the second index!
+                        stepIndex = [kUserProgressStepIndex integerValue];
+                        
+                        break;
+                    }
+                    case 2:
+                    {
+                        stepIndex = DepthTwoResult(step, result);
+                        
+                        break;
+                    }
+                    case 3:
+                    {
+                        stepIndex = DepthThreeResult(step, result);
+                        
+                        break;
+                    }
+                    case 4:
+                    {
+                        stepIndex = kNegativeStepIndex;
+                        
+                        break;
+                    }
+                        
+                    default:
+                    {
+                        stepIndex = kNegativeStepIndex;
+                        
+                        break;
+                    }
+                }
+            }
+            else
+            {
+                stepIndex = kNegativeStepIndex;
+            }
+        }
+        
+        return stepIndex;
+    };
+    
+    stepIndex = DetermineNextStep(step, result);
+    
+    if (stepIndex < 0)
+    {
+        step = nil;
+    }
+    else
+    {
+        step = self.steps[stepIndex];
+    }
+    
+    return step;
+}
+
+- (nullable ORKStep *)stepBeforeStep:(nullable ORKStep *)step withResult:(ORKTaskResult *) __unused result
+{
+    NSInteger stepIndex = [kInstructionStepIndex integerValue];
+
+    if ([step.identifier isEqualToString:kInstructionStepID])
```

Comment:
@jwarmkessel: [here](https://github.com/rsanchezsaez/RKNavigableTaskExample) you can have a look on how this exact task can be achieved using the recently added `ORKNavigableOrderedTask` class and defining a few simple rules.

You get to avoid custom `stepBeforeStep:` and `stepAfterStep:` implementations. You also avoid having to define the fake duplicate step `kReasonsTwoStepID` for tracking the correct previous step.